spektrum/runner.py

```diff
@@ -92,21 +92,48 @@ def run(self, search_paths, module_name=None, metadata=None, test_names=None, ex
 
         return self.reporting.success
 
+    def _map_classes(self, classes, class_dict, parent):
+        for cls in classes:
+            cur_node = f'{parent}.{(cls.__name__)}'
+            class_dict[cur_node] = cls
+            children = find_children(cls)
+            self._map_classes(children, class_dict, cur_node)
+
     def filter_by_module_name(self, classes, name):
         found = [
             cls
             for cls in classes
             if name in '{}.{}'.format(cls.__module__, cls.__name__)
         ]
 
-        # Only search children if the class cannot be found at the package lvl
         if not found:
-            children = [find_children(cls) for cls in classes]
-            found = [
-                cls
-                for cls in itertools.chain.from_iterable(children)
-                if name in '{}.{}'.format(cls.__module__, cls.__name__)
-            ]
+            try:
+                module = next(
+                    cls
+                    for cls in classes
+                    if cls.__module__ in name
+                )
+                class_dict = {}
+                self._map_classes([module], class_dict, module.__module__)
+
+                found = {}
+
+                def parent_exists(name, class_map):
+                    for cls_name in class_map.keys():
+                        if cls_name in name:
+                            return True
+                    return False
+
+                if '*' in name:
+                    for cls_name, cls in class_dict.items():
+                        if re.search(f'{name}$', cls_name):
+                            if not parent_exists(cls_name, found):
+                                found[cls_name] = cls
+                    found = [cls for cls in found.values()]
```

Comment:
We should use two separate variables for a dict and a list. Not really fond of redefining variable types... I know we do it occasionally but we should try not to. It just opens up the door for potential issues.